src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImpl.java

```diff
@@ -78,9 +85,13 @@ public HarvestServiceImpl(HttpHarvester harvester,
 
       recordHeaderIterator.forEach(recordHeader -> {
         try {
-          var oaiRecord = harvesterOai
-              .harvestRecord(oaiRepository, recordHeader.getOaiIdentifier());
-          records.add(new ByteArrayInputStream(oaiRecord.getRecord().readAllBytes()));
+          if(records.size() == maxRecords){
```

Comment:
After having into a look into the code of metis, this implementation seems to be the only way to make this work.
I understand the problem, and indeed, in the future this could cause performance issues, this is something we need to fix on metis and then apply it here.

Reply:
I actually managed to improve the OAI harvesting code in metis-sandbox, it was not needed to change anything in metis related to OAI harveting. It still takes a few minutes to reach the maximum number of harvested records